dask/dataframe/io/parquet/core.py

```diff
@@ -911,6 +912,18 @@ def get_engine(engine):
         return eng
 
     elif engine in ("pyarrow", "arrow", "pyarrow-legacy", "pyarrow-dataset"):
+
+        pa = import_required("pyarrow", "`pyarrow` not installed")
+        pa_version = parse_version(pa.__version__)
+
+        if engine in ("pyarrow", "arrow"):
+            engine = "pyarrow-dataset"
+        elif pa_version.major >= 5 and engine == "pyarrow-legacy":
+            raise ValueError(
```

Comment:
Maybe make this a warning instead of an error? (at least on the short term) If people run into issues because of this switch in defaults, I think it will be useful that they can still quickly check with "pyarrow-legacy" explicitly if the issue is caused by the switch in engine or not.